src/main/java/xyz/kyngs/herbot/handlers/user/UserProfile.java

```diff
@@ -0,0 +1,77 @@
+package xyz.kyngs.herbot.handlers.user;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class UserProfile {
+
+    private final List<String> perms;
+    private long coins;
+    private long xp;
+    private int dailyStreak;
+    private LocalDate lastDailyClaim;
+
+    public UserProfile(long coins, long xp, int dailyStreak, LocalDate lastDailyClaim, List<String> perms) {
+        this.coins = coins;
+        this.xp = xp;
+        this.dailyStreak = dailyStreak;
+        this.lastDailyClaim = lastDailyClaim;
+        this.perms = perms;
+    }
+
+    public boolean hasPermission(String perm) {
+        return perm.contentEquals("") || perms.contains("*") || perms.contains(perm);
```

Comment:
So user has permission, when perm = ""? This doesn't look like something we want.

Reply:
If command has permission set to "" it means that it does not require any permission, therefore User#hasPermission will return true everytime the supplied permission is "'